src/kv/test/kv_serialisation.cpp

```diff
@@ -291,6 +382,137 @@ TEST_CASE(
   }
 }
 
+struct CustomClass2
+{
+  std::string s;
+  size_t n;
+};
+
+struct CustomJsonSerialiser
+{
+  using Bytes = kv::experimental::SerialisedRep;
+
+  static Bytes to_serialised(const CustomClass2& c)
+  {
+    nlohmann::json j = nlohmann::json::object();
+    j["s"] = c.s;
+    j["n"] = c.n;
+    const auto s = j.dump();
+    return Bytes(s.begin(), s.end());
+  }
+
+  static CustomClass2 from_serialised(const Bytes& b)
+  {
+    const auto j = nlohmann::json::parse(b);
+    CustomClass2 c;
+    c.s = j["s"];
+    c.n = j["n"];
+    return c;
+  }
+};
+
+struct KPrefix
+{
+  static constexpr auto prefix = "This is a key:";
+};
+
+struct VPrefix
+{
+  static constexpr auto prefix = "Here follows a value:";
+};
+
+template <typename T>
+struct CustomVerboseDumbSerialiser
```

Comment:
This is a bit over-complicated for test scaffolding. It is supposed to show both "you can serialise however you want" and "you can serialise your Ks and Vs differently, even when they're the same type", but it also shows "you can serialise in really stupid formats if you want".